src/main/java/org/folio/cql2rmapi/CQLParserForRMAPI.java

```diff
@@ -26,319 +13,86 @@
-
-  private static final String ERROR = "Unsupported Query Format : ";
-  private static final String UNSUPPORTED = " is not supported.";
-  private static final String CQL_SERVER_CHOICE = "cql.serverChoice";
-  private static final String RM_API_TITLE = "titlename";
-  private static final String SOURCE = "source";
-  private static final String CODEX_SOURCE = "codex.source";
-  private static final String SELECTED = "ext.selected";
-  private static final String TITLE = "title";
-  private static final String CODEX_TITLE = "codex.title";
-  private static final String RESOURCE_TYPE = "resourceType";
-  private static final String CODEX_RESOURCE_TYPE = "codex.resourceType";
-  private static final String IDENTIFIER = "identifier";
-  private static final String CODEX_IDENTIFIER = "codex.identifier";
-  private static final String SUBJECT = "subject";
-  private static final String CODEX_SUBJECT = "codex.subject";
-  private static final String PUBLISHER = "publisher";
-  private static final String CODEX_PUBLISHER = "codex.publisher";
-  private static final String ID = "id";
-  private static final String CODEX_ID = "codex.id";
-
   private static final String SELECTION_QUERY_PARAM = "&selection=";
 
-  String searchField;
-  String searchValue;
-  String filterType;
-  String filterValue;
-  String sortType;
-  String selection;
-  int countRMAPI;
-  int instanceIndex;
-  int instanceLimit;
-  private boolean idSearchField;
-  private String idSearchFieldValue;
-  List<String> queriesForRMAPI = new ArrayList<>();
-
-
-  private enum validSources {
-    ALL, KB
-  }
-
-  public CQLParserForRMAPI(String query, int offset, int limit) throws QueryValidationException, UnsupportedEncodingException {
-    if(limit != 0 && query != null) {
-      final CQLNode node = initCQLParser(query);
-      checkNodeInstance(node);
-      //If it is an id search field, we do not need to build the query since we can directly invoke RM API to look for that id
-      if(!idSearchField) {
-        instanceLimit = limit;
-        int pageOffsetRMAPI = computePageOffsetForRMAPI(offset, limit);
-        queriesForRMAPI.add(buildRMAPIQuery(limit, pageOffsetRMAPI));
-        instanceIndex = computeInstanceIndex(offset, limit);
-        while (checkIfSecondQueryIsNeeded(offset, limit, pageOffsetRMAPI)) {
-          queriesForRMAPI.add(buildRMAPIQuery(limit, ++pageOffsetRMAPI));
-        }
-      }
-    } else {
-      throw new QueryValidationException(ERROR + "Limit/Query suggests that no results need to be returned.");
-    }
-  }
-
-  CQLNode initCQLParser(String query) throws QueryValidationException {
-    final CQLParser parser = new CQLParser();
-    try {
-      return parser.parse(query);
-    } catch (CQLParseException | IOException e) {
-      throw new QueryValidationException(ERROR + "Search query is in an unsupported format.", e);
+  private int instanceIndex;
+  private int instanceLimit;
+  private List<String> queriesForRMAPI = new ArrayList<>();
```

Comment:
Really, really minor comment -- since instance is one of the codex types -- instanceIndex and instanceLimit seems like they are specific to the instance type (even though it is a common) -- wonder if another name (like itemLimit or itemIndex could be used)